tests/test_pay.py

```diff
@@ -5420,6 +5420,51 @@ def test_invoice_pay_desc_with_quotes(node_factory):
     l1.rpc.pay(invoice, description=description)
 
 
+@pytest.mark.xfail(strict=True)
+def test_self_sendpay(node_factory):
```

Comment:
This test looks like a copy of another more generic one. I think we should be more specific for this one, and just test a single self-payment, otherwise we are just increasing the surface that has to be reworked if a change happens, and without any additional benefits, other than tests that are more likely to run into flakes.

Reply:
Almost.  It's awkardly different!  We return much better errors for selfpay than for remotepay, since we actually know things like "that invoice has expired" or "you used the wrong secret" instead of a simple `WIRE_INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS` failure code.

Reply:
I've added a comment about this at the top of the test...